packages/gateway/src/ipfs.js

```diff
@@ -8,7 +8,7 @@ import { normalizeCid } from './utils/cid.js'
  */
 export async function ipfsGet(request, env) {
   const cid = request.params.cid
-  const path = request.url.split(cid)[1] || ''
+  const path = request.url.split(`/ipfs/${cid}`)[1] || ''
```

Comment:
Can we instead be more explicit about what part of the URL we're considering? 

```js
new URL(request.url).pathname.split(cid)[1]
```

There's nothing to say the reason for the longer matcher string is because the CID could appear in the domain of the URL. Arguably using `pathname` is no better...perhaps a comment here could help.

...also what if the CID happens to appear in the path multiple times?

...can we also preserve querystring?